aidial_adapter_bedrock/llm/model/claude/v3/converters.py

```diff
@@ -191,7 +191,7 @@ async def to_claude_messages(
                 )
             case _:
                 raise ValidationError(
-                    f"Unknown type of of message! {type(message)}"
+                    f"Not supported type of message! {type(message)}"
```

Comment:
The error message is misleading, since these `*Function*` cases are in fact supported, but they were eliminated in 
 `emulate_function_message` before reaching this method.

Let's keep logic transforming different cases of the same type together - move two cases from `emulate_function_message` to this method.

Reply:
I don't think merging this two functions is good idea, but I agree, this error is misleading

Changed this part to:
```python
        case AIFunctionCallMessage() | HumanFunctionResultMessage():
            raise ValueError(
                "Function messages should be converted to tool messages before sending to Claude!"
            )
```

It will be similar to error above for SystemMessages
```python
           case SystemMessage():
               raise ValueError(
                   "System message is only allowed as the first message"
               )

Reply:
Let's suppose  (a)`tools_mode==TOOLS` and (b) there is a function message in the list of messages. 

The error `Function messages should be converted to tool messages before sending to Claude!` will be thrown and it will be misleading. Not only the message of it is misleading, but also the type - it's internal `ValueError` _(which results in 500 response)_, but would rather be `ValidationError` _(which results in 400 response)_.

I suggest to modify `convert_function_message` in the following way:
* call it unconditionally even for `tools_mode==TOOLS`
* check the consistency of messages with the given `tools_mode` inside the method
* do the conversion to tool messages (no changes here)
* modify the types of output message to reflect that certain types of messages were eliminated:

```python
def convert_function_message(
    tools_mode: ToolsMode,
    message: BaseMessage | ToolMessage,
) -> BaseMessage | HumanToolResultMessage | AIToolCallMessage:
    ...
```

This will allow you to remove `Function messages should be converted to tool messages before sending to Claude!` error altogether, because it's guaranteed by types it will never be hit:

```python
async def to_claude_messages(
    messages: List[BaseMessage | HumanToolResultMessage | AIToolCallMessage],
    file_storage: Optional[FileStorage],
) -> Tuple[Optional[str], List[MessageParam]]:
    ...
```

Reply:
Fixed, also renamed `convert_function_messages` to `process_with_tools`, since it's much more validation there, not only conversion